### ISSUE TITLE
Event handler's arguments can be changed to 'in' (C# 7.2) without a compilation error, leading to runtime crashes

### ISSUE THREAD

**Issue by 0xd4d** [dotnet/roslyn#23896]
Version: Microsoft (R) Visual C# Compiler version 2.6.0.62329 (5429b35d)

Expected: Some type of error should be generated when trying to add event handlers `Program_Event1b` and `Program_Event2b`

Actual: The compiler generates an EXE that crashes at runtime.

Example:

```C#
using System;

namespace ConsoleApp1 {
	class Program {
		readonly struct EventArgs1 {
			public int Value { get; }
			public EventArgs1(int value) => Value = value;
		}
		sealed class EventArgs2 {
			public int Value { get; }
			public EventArgs2(int value) => Value = value;
		}
		static event EventHandler<EventArgs1> Event1;
		static event EventHandler<EventArgs2> Event2;
		static void Main(string[] args) {
			Event1 += Program_Event1a;
			Event1 += Program_Event1b;
			Event2 += Program_Event2a;
			Event2 += Program_Event2b;

			try {
				Event1?.Invoke(1111.1111, new EventArgs1(1111));
			}
			catch (Exception ex) {
				Console.WriteLine($"EX: {ex.GetType().FullName}");
			}

			try {
				Event2?.Invoke(2222.2222, new EventArgs2(2222));
			}
			catch (Exception ex) {
				Console.WriteLine($"EX: {ex.GetType().FullName}");
			}
		}

		static void Program_Event1a(object sender, EventArgs1 e) => Console.WriteLine($"Event1: sender={sender}, e.Value={e.Value}");
		static void Program_Event2a(object sender, EventArgs2 e) => Console.WriteLine($"Event2: sender={sender}, e.Value={e.Value}");
		static void Program_Event1b(in object sender, in EventArgs1 e) => Console.WriteLine($"Event1: sender={sender}, e.Value={e.Value}");
		static void Program_Event2b(in object sender, in EventArgs2 e) => Console.WriteLine($"Event2: sender={sender}, e.Value={e.Value}");
	}
}

```

**Comment by ufcpp** [comment]
essentially dup of https://github.com/dotnet/roslyn/issues/23636#issuecomment-350544058

**Comment by 0xd4d** [comment]
Yes, this seems to be fixed by #23397
